index.js

```diff
@@ -202,8 +202,18 @@
       var name = 'fantasy-land/' + _name;
       return requirements[_name] === Constructor ?
         function(typeRep) {
-          return funcPath([name], typeRep) ||
-                 implPath([/function (\w*)/.exec(typeRep)[1], name]);
+          var f = funcPath([name], typeRep);
+          if (f == null && typeof typeRep === 'function') {
+            /* istanbul ignore next */
+            var functionName = 'name' in function() {} ?
+                function functionName(f) { return f.name; } :
+                function functionName(f) {
+                  var match = /function (\w*)/.exec(f);
+                  return match && match[1];
+                };
```

Comment:
I believe that is the case (if you don't count es2015+ -> es5 transpilation). But it's more about preferring invariants. I suppose I could just do a check on the return value and only place it in the array if it's a string.